sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -176,15 +175,15 @@ def model(self):
     
     def _deserialize_data(self, event_type):
         """
-        Sets self._model.data to strongly typed event object if event type exists in _event_type_mappings.
+        Sets self._model.data to strongly typed event object if event type exists in _event_mappings.
         Otherwise, sets self._model.data to None.
 
         :param str event_type: The event_type of the EventGridEvent object or the type of the CloudEvent object.
         """
         # if system event type defined, set model.data to system event object
-        if event_type in DeserializedEvent._event_type_mappings:
-            self._model.data = (DeserializedEvent._event_type_mappings[event_type]).deserialize(self._model.data)
-        else:    # else, if custom event, then model.data is dict and should be set to None
+        try:
+            self._model.data = (_event_mappings[event_type]).deserialize(self._model.data)
+        except KeyError:
```

Comment:
```suggestion
        except KeyError: # else, if custom event, then model.data is dict and should be set to None
```

your comment was useful and I heartily support commenting :)